extensions/register.md

```diff
@@ -0,0 +1,157 @@
+---
+title: The REGISTER command
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Ed Kellett"
+    period: "2020"
+    email: "e@kellett.im"
+  -
+    name: "Val Lorentz"
+    period: "2021"
+    email: "progval+ircv3@progval.net"
+---
+
+
+## Motivation
+
+The lack of a standardized way to register an account has been a source of frustration for many participants
+for several years.
+
+This isn't the first attempt to fix it. Unlike the withdrawn `ACC` proposal, this effort aims to be more
+appealing to implementors by being severely restricted in scope. The only register-verify flows supported
+are those already in use in the wild. The authors hope that this stripped-down solution will serve as the
+basis for future organic expansion.
+
+### Capability
+
+This specification adds the `draft/register` capability, whose presence signifies that the server
+accepts the `REGISTER` command.
+
+The capability has an optional value, a comma-separated list of key-value pairs; the format is intended
+to follow the precedent set by [the multiline draft][multiline].
+
+The defined keys are:
+
+ * `before-connect` - if present, indicates the server supports early registration, so it
+   MUST NOT use `FAIL REGISTER COMPLETE_CONNECTION_REQUIRED`
+ * `email-required` - if present, registrations require a valid email address to process
```

Comment:
Should there be a fail code for when the user sends * instead of an email or is INVALID_EMAIL enough?

Reply:
INVALID_EMAIL should be enough, the client software can work it out based on that and the message should tell the user what went wrong